app/blueprints/assessments/routes.py

```diff
@@ -411,36 +362,493 @@ def get_sorted_application_overviews(application_overviews, column, reverse=Fals
+        "is_expression_of_interest": round.is_expression_of_interest,
+    }
+
+    all_applications_metadata = get_application_overviews(fund.id, round.id, "")
+
+    unfiltered_stats = process_assessments_stats(all_applications_metadata)
+
+    return render_template(
+        "assessor_type.html",
+        selected_assessments=selected_assessments,
+        selected_assessor_role=selected_assessor_role,
         round_details=round_details,
-        query_params=search_params,
-        asset_types=asset_types,
-        funding_types=funding_types,
-        cohort=cohort,
-        assessment_statuses=assessment_statuses,
-        joint_application_options=joint_application_options,
-        show_clear_filters=show_clear_filters,
         stats=unfiltered_stats,
-        team_flag_stats=teams_flag_stats,
-        is_active_status=is_active_status,
-        sort_column=sort_column,
-        sort_order=sort_order,
-        tag_option_groups=tag_option_groups,
-        tags=tags_in_application_map,
-        tagging_purpose_config=Config.TAGGING_PURPOSE_CONFIG,
-        countries=all_application_locations.countries,
-        regions=all_application_locations.regions,
-        local_authorities=all_application_locations._local_authorities,
-        migration_banner_enabled=Config.MIGRATION_BANNER_ENABLED,
-        dpi_filters=dpi_filters,
-        users=["All", "Not assigned"] + fund_user_aliases,
+        form=form,
+    )
+
+
+@assessment_bp.route(
+    "/assign/<fund_short_name>/<round_short_name>/assessors",
+    methods=["POST"],
+)
+@check_access_fund_short_name_round_sn
+def assessor_type_list(fund_short_name: str, round_short_name: str):
+    selected_assessments = request.form.getlist("selected_assessments")
+    assessor_role = request.form.getlist("assessor_role")[0]
+    selected_users = request.form.getlist("selected_users")
+
+    form = AssessorChoiceForm()
+
+    if form.validate_on_submit():
+        return redirect(
+            url_for(
+                "assessment_bp.assignment_overview",
+                fund_short_name=fund_short_name,
+                round_short_name=round_short_name,
+            ),
+            307,
+        )
+
+    fund = get_fund(
+        fund_short_name,
+        use_short_name=True,
+        ttl_hash=get_ttl_hash(Config.LRU_CACHE_TIME),
+    )
+    if not fund:
+        return redirect("/assess/assessor_tool_dashboard/")
+
+    round = get_round(
+        fund_short_name,
+        round_short_name,
+        use_short_name=True,
+        ttl_hash=get_ttl_hash(Config.LRU_CACHE_TIME),
+    )
+
+    round_details = {
+        "assessment_deadline": round.assessment_deadline,
+        "round_title": round.title,
+        "fund_name": fund.name,
+        "fund_short_name": fund_short_name,
+        "round_short_name": round_short_name,
+        "is_expression_of_interest": round.is_expression_of_interest,
+    }
+
+    thread_executor = ContextAwareExecutor(
+        max_workers=4,
```

Comment:
shall  we add this into the config ? and for every request are we going to create this ContextAwareExecutor ? better to have one executor for entire app other wise more ContextAwareExecutor will be created

Reply:
So we don't use the ContextAwareExecutor for every request, only those that involve a lot of calls to the stores. The problem with having a single executor for the app is that all requests can/will use it, and that will introduce coupling between requests ... so if we have one bad request that blocks the executor, it may end up blocking other unrelated requests. It would also be a nightmare to debug. 

I think creating an executor per request is probably the safest approach (and it will be destroyed at the end of the request anyway).

(To be honest, I don't really like using the ContextAwareExecutor approach ... but we have to because of the microservice architecture )

Reply:
shall we have a call about this I have some previous knowledge about java it caused an server overload but since we have ecs it can handle it but it creates unnecessary overload

Reply:
Summary of call : `ContextAwareExecutor` doesn't automatically clean up the threads when it goes out of scope, so it is necessary to call `ContextAwareExecutor.executor.shutdown()` to ensure the necessary clean up takes place.